src/target/riscv/riscv.c

```diff
@@ -5588,6 +5588,48 @@ COMMAND_HANDLER(handle_riscv_virt2phys_mode)
 	return ERROR_OK;
 }
 
+COMMAND_HANDLER(riscv_smp_add_ext_triggers)
+{
+	struct target *target = get_current_target(CMD_CTX);
+	RISCV_INFO(r);
+
+	if (CMD_ARGC % 2) {
+		LOG_ERROR("Command takes an even number of parameters.");
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
+
+	if (CMD_ARGC > 2 * RISCV_MAX_EXTTRIGGERS) {
+		LOG_ERROR("The number of parameters can not exceeds %d.", 2 * RISCV_MAX_EXTTRIGGERS);
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
+
+	unsigned int index = 0;
```

Comment:
The command is called "**add**_ext_triggers" so the user may think that it can be called multiple times and the parameters would merge. 

This is not the case - the function overwrites the values of the previous call. 

Please pick one of these two options:

- Rename the function to "set" 
- Or make sure that the previously set triggers are not overwritten. In this case the function can be simplified to take just two arguments, and the user can call it as many times as they need.

Reply:
I am afraid this is not fully addressed yet. If the user calls these two commands:

```
riscv smp_set_ext_triggers halt_group 2 halt_group 3 halt_group 4
riscv smp_set_ext_triggers halt_group 5
```

... the current result would be that only triggers 5, 3 and 4, would be assigned. It would not be very intuitive.

I would prefer if all the triggers (2, 3, 4 and 5) were assigned.

When the trigger structure (`struct riscv_ext_trigger`) is re-arranged as suggested in another comment, this should not be a problem anymore.